core/src/Cuda/Kokkos_Cuda_Task.hpp

```diff
@@ -104,7 +104,9 @@ class TaskQueueSpecialization<SimpleTaskScheduler<Kokkos::Cuda, QueueType>> {
     extern __shared__ int32_t shmem_all[];
 
     int32_t* const warp_shmem =
-        shmem_all + (threadIdx.z * shmem_per_warp) / sizeof(int32_t);
+        shmem_all +
+        (static_cast<std::uintptr_t>(threadIdx.z) * shmem_per_warp) /
```

Comment:
Is `std::uintptr_t` the expected type here ? I thought it was more meant to be used for conversions between pointers and integers ? In this case I would expect `std::size_t` to be more suitable.

Reply:
This does pointer arithmetic `shmem_all` is also a pointer after all and is also used as such in the next line.

Reply:
I understand that `shmem_all` is a pointer but this is usual arithmetic pointer right ? I feel `std::size_t` should be enough, the type `std::uintptr_t` is not required by the standard.

I will not be blocking for that because I think it works, though.

Reply:
I think it doesn't matter much in the end.